core/controllers/base_test.py

```diff
@@ -274,6 +288,132 @@ def test_renders_error_page_with_iframed(self):
             'Uh-oh! The Oppia exploration you requested may have been removed '
             'or deleted.', response.body)
 
+    def test_dev_mode_cannot_be_true_on_production(self):
+        del sys.modules['feconf']
+        server_software_swap = self.swap(
+            os, 'environ', {'SERVER_SOFTWARE': 'Production'})
+        assert_raises_regexp_context_manager = self.assertRaisesRegexp(
+            Exception, 'DEV_MODE can\'t be true on production.')
+        with assert_raises_regexp_context_manager, server_software_swap:
+            import feconf  # pylint: disable-all
+
+    def test_valid_pillow_path(self):
+        import appengine_config
+        assert_raises_regexp_context_manager = self.assertRaisesRegexp(
+            Exception, 'Invalid path for oppia_tools library: invalid_path')
+
+        def mock_os_path_join_for_pillow(*args):
+            path = ''
+            if args[1] == 'Pillow-6.0.0':
+                return 'invalid_path'
+            else:
+                for arg in args:
+                    path += arg + '/'
+                return path
+
+        pil_path_swap = self.swap(os.path, 'join', mock_os_path_join_for_pillow)
+        del sys.modules['appengine_config']
+
+        with assert_raises_regexp_context_manager, pil_path_swap:
+            import appengine_config  # pylint: disable-all
```

Comment:
I added this pragma since pylint was throwing a bunch of errors on reimporting modules

Reply:
Which errors? Please always provide a trace.

Reply:
The error messages are `reimported`, `unused-variable`, `redefined-outer-name`, and `unused-import`. These errors are for all the re-imports in this file. I am just re-importing the modules to run them under different conditions. Also for re-importing, I need to do `del sys.modules['module_name']`

Reply:
Please add comments inline to explain this for the benefit of future devs. (Ditto for the previous test.) Do this everywhere you have the pragma, and everywhere you have the `del sys.modules`.

In general, if you think a comment is useful for a reviewer, it is probably also useful for other devs looking at this file in the future.

Reply:
Done